components/backup-stream/src/endpoint.rs

```diff
@@ -212,10 +214,33 @@ where
     RT: RaftStoreRouter<E> + 'static,
     PDC: PdClient + 'static,
 {
-    fn on_fatal_error(&self, _task: String, _err: Box<Error>) {
-        // This is a stub.
-        // TODO: implement the feature of reporting fatal error to the meta storage,
-        //       and pause the task then.
+    fn get_meta_client(&self) -> MetadataClient<S> {
+        self.meta_client.as_ref().unwrap().clone()
+    }
+
+    fn on_fatal_error(&self, task: String, err: Box<Error>) {
+        // Let's pause the task locally first.
```

Comment:
We will need another PR about `meta_client.pause(taskname)` later.

Reply:
done